lib/tracking/trackProduct.js

```diff
@@ -1,43 +1,37 @@
-import track from "./track";
 import getProductTrackingData from "./utils/getProductTrackingData";
 import getVariantTrackingData from "./utils/getVariantTrackingData";
 
 /**
- * trackProduct HOC tracks a product
+ * trackProduct to tracks a product
  * @name trackProduct
- * @param {Object} options options to supply to tracking HOC
+ * @param {Object} payload data to be tracked
+ * @param {Object} router router next js
  * @returns {React.Component} - component
  */
-export default (options) => (
-  track(({ product, router }, state, functionArgs) => {
-    let data = {};
-    const { variant, optionId, action } = (functionArgs && functionArgs[0]) || [];
+export default (payload, router) => {
+  const { product, variant, optionId, action } = payload;
+  let data = {};
+  // If product data is provided as a prop, then process the data for tracking
+  if (product) {
+    data = {
+      action,
+      ...getProductTrackingData(product, router)
+    };
 
-    // If product data is provided as a prop, then process the data for tracking
-    if (product) {
+    // Add variant data if available
+    if (variant) {
       data = {
-        action,
-        ...getProductTrackingData(product)
+        ...data,
+        ...getVariantTrackingData({
+          product,
+          router,
+          variant, // Object representing a variant. (Required)
+          optionId // Selected option of the provided variant, if available. (Optional)
+        }),
+        allDataProduct: product // Full product document for additional data. (Optional)
```

Comment:
I'd imagine this to possibly get quite a big payload - is this picked up anywhere?

Reply:
What I mean is, if `allDataProduct` is sent alongside the event.
The full product could potentially be quite big, too big to send it with each event.
`trackProduct` would currently pick some relevant fields only and return only that data.
So I was wondering if the tracking providers currently pick up the content of `allDataProduct` (which would be bad) or if it is filtered out somewhere in the process, I didn‘t look that up yet.
If there is no other usecase of it, I‘d just want to remove it.

Reply:
Got it